lib/compress/zstd_compress.c

```diff
@@ -2586,12 +2586,27 @@ size_t ZSTD_mergeBlockDelimiters(ZSTD_Sequence* sequences, size_t seqsSize) {
     return out;
 }
 
-/* Returns true if the given block is a RLE block */
-static int ZSTD_isRLE(const BYTE *ip, size_t length) {
+/* Unrolled loop to read four size_ts of input at a time. Returns 1 if is RLE, 0 if not. */
+static int ZSTD_isRLE(const BYTE* src, size_t length) {
+    const BYTE* ip = src;
+    const BYTE value = ip[0];
+    const size_t valueST = value * 0x0101010101010101ULL;
```

Comment:
You'll need to make this an explicit cast to silence the visual warnings.
```suggestion
    const size_t valueST = (size_t)((U64)value * 0x0101010101010101ULL));
```